proto/v1/external_vcs_service.proto

```diff
@@ -0,0 +1,202 @@
+    option (google.api.http) = {
+      post: "/v1/external_vcs"
+      body: "external_vcs"
+    };
+    option (google.api.method_signature) = "";
+  }
+
+  rpc GetExternalVCS(GetExternalVCSRequest) returns (ExternalVCS) {
+    option (google.api.http) = {
+      get: "/v1/{name=external_vcs/*}"
+    };
+    option (google.api.method_signature) = "name";
+  }
+
+  rpc ListExternalVCS(ListExternalVCSRequest) returns (ListExternalVCSResponse) {
+    option (google.api.http) = {
+      get: "/v1/external_vcs"
+    };
+    option (google.api.method_signature) = "";
+  }
+
+  rpc UpdateExternalVCS(UpdateExternalVCSRequest) returns (ExternalVCS) {
+    option (google.api.http) = {
+      patch: "/v1/{external_vcs.name=external_vcs/*}"
+      body: "external_vcs"
+    };
+    option (google.api.method_signature) = "external_vcs,update_mask";
+  }
+
+  rpc DeleteExternalVCS(DeleteExternalVCSRequest) returns (google.protobuf.Empty) {
+    option (google.api.http) = {
+      delete: "/v1/{name=external_vcs/*}"
+    };
+    option (google.api.method_signature) = "name";
+  }
+
+  rpc ListExternalRepositories(ListExternalRepositoriesRequest) returns (ListExternalRepositoriesResponse) {
+    option (google.api.http) = {
+      get: "/v1/{name=external_vcs/*}/external_repositories"
+    };
+    option (google.api.method_signature) = "name";
+  }
+}
+
+message CreateExternalVCSRequest {
+  ExternalVCS external_vcs = 1 [(google.api.field_behavior) = REQUIRED];
+}
+
+message GetExternalVCSRequest {
+    // The name of the external VCS to retrieve.
+    // Format: external_vcs/{external_vcs}
+    string name = 1 [(google.api.field_behavior) = REQUIRED];
+}
+
+message ListExternalVCSRequest {
+  // Not used. The maximum number of external VCS to return. The service may return fewer than this value.
+  // If unspecified, at most 100 external VCS will be returned.
+  // The maximum value is 1000; values above 1000 will be coerced to 1000.
+  int32 page_size = 1;
+    
+  // Not used. A page token, received from a previous `ListExternalVCS` call.
+  // Provide this to retrieve the subsequent page.
+  string page_token = 2;
+}
+
+message ListExternalVCSResponse {
+  // The list of external VCS.
+  repeated ExternalVCS external_vcs = 1;
+    
+  // Not used. A token, which can be sent as `page_token` to retrieve the next page.
+  // If this field is omitted, there are no subsequent pages.
+  string next_page_token = 2;
+}
+
+message UpdateExternalVCSRequest {
+  ExternalVCS external_vcs = 1 [(google.api.field_behavior) = REQUIRED];
+    
+  // The list of fields to be updated.
+  google.protobuf.FieldMask update_mask = 2 [(google.api.field_behavior) = REQUIRED];
+}
+
+message DeleteExternalVCSRequest {
+  // The name of the external VCS to delete.
+  // Format: external_vcs/{external_vcs}
+  string name = 1 [(google.api.field_behavior) = REQUIRED];
+}
+
+message ListExternalRepositoriesRequest {
```

Comment:
How is it used?

Reply:
Frontend should list the external repository when user convert the project to gitops mode.